module/zfs/zfs_rlock.c

```diff
@@ -106,10 +106,15 @@ zfs_range_lock_writer(znode_t *zp, rl_t *new)
 	avl_tree_t *tree = &zp->z_range_avl;
 	rl_t *rl;
 	avl_index_t where;
-	uint64_t end_size;
 	uint64_t off = new->r_off;
 	uint64_t len = new->r_len;
 
+#ifdef _KERNEL
+	uint64_t end_size;
+#else
+	ASSERT3U(zp->z_is_zvol, ==, B_TRUE);
+#endif
```

Comment:
It's not clear it me it's worth the bother to add the `z_is_zvol` member if it's only ever used for this assertion.  Also wouldn't it make more sense to set `z_is_zvol = 0` and then initialize `z_size` and `z_blksz` to reasonable values so the `if (!zp->z_is_zvol) { }` wouldn't need to be disabled?